src/model.py

```diff
@@ -807,21 +801,30 @@ def init_optimizer(self):
         
         for task in self.subtasks:  # TODO make sure in self.components in others
             lr = task_lrs[task] if task_lrs.get(task) is not None else self.learning_rate
-            optimizers[task] = optimizer(
-                    self.bert.parameters(),  # NOTE all tasks can optimize bert params if need
-                    lr=self.learning_rate  # use main learning rate for bert training
-            ) # TODO test other optimizers?
 
-            if "shared" in self.components.keys():
-                for layer in self.components["shared"]:
+            for i, layer in enumerate(self.components[task]):
+                if i == 0:
+                    optimizers[task] = optimizer(
+                        self.components[task][layer].parameters(),
+                        lr=lr  # use main learning rate for bert training
+                    ) # TODO test other optimizers?
+                    
+                    if "shared" in self.components.keys():
+                        for layer in self.components["shared"]:
+                            optimizers[task].add_param_group(
+                                {"params": self.components["shared"][layer].parameters(), "lr":lr}
+                            )
+
+                    if self.find(f"bert_{task}", default=False) and self.finetune:
+                        optimizers[task].add_param_group(
+                            {"params": self.bert.parameters(), "lr":self.learning_rate}
+                        )
+
+                else:
```

Comment:
Not every task trains bert by default anymore. Now configurable via model param `bert_{task}`, as long as `bert_finetune=True` of course. 

NOTE: Should probably allow flexibility so that `bert_finetune` doesn't need to be called.